sw/device/lib/base/BUILD

```diff
@@ -0,0 +1,48 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "base",
+    copts = [
+        "-fno-builtin",
+    ],
+    srcs = [
+        "bitfield.c",
+        "hardened.c",
+        "memory.c",
+    ],
+    hdrs = [
+        "bitfield.h",
+        "csr.h",
+        "csr_registers.h",
+        "hardened.h",
+        "macros.h",
+        "memory.h",
+        "mmio.h",
+        "stdasm.h",
+    ],
+)
+
+
+cc_library(
+    name = "mmio",
+    srcs = [
+        "mmio.c",
+    ],
+    hdrs = [
+        "mmio.h",
+    ],
+    linkstatic = True,
```

Comment:
You shouldn't need this: static linking will be enforced by cc_binary rules which include this as a dependency.